src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientsService.kt

```diff
@@ -42,24 +43,42 @@ class RestrictedPatientsService(
       commentText = dischargeToHospital.commentText
     )
 
-    return transform(restrictedPatientsRepository.save(restrictedPatient))
+    return transform(
+      restrictedPatientsRepository.save(restrictedPatient),
+      response?.restrictivePatient?.supportingPrison,
+      response?.restrictivePatient?.dischargedHospital,
+      response?.restrictivePatient?.supportingPrison
+    )
   }
 
   fun getRestrictedPatient(prisonerNumber: String): RestrictedPatientDto {
     val restrictedPatient = restrictedPatientsRepository.findByPrisonerNumberAndActiveTrue(prisonerNumber)
       ?: throw EntityNotFoundException("No restricted patient record found for prison number $prisonerNumber")
 
-    return transform(restrictedPatient)
+    val agencies = prisonApiGateway.getAgencyLocationsByType("INST")
```

Comment:
I think these three calls need to be cached.

The information changes very rarely and is called on a change of data (albeit just restricted patients).